File: libs/power-log-parser/src/lib/parsers/utils/bgs-utils.ts
import { CardIds } from '@firestone-hs/reference-data';

export class BgsUtils {
	static IsBaconGhost(cardId: string): boolean {
		return (
			cardId === CardIds.LadyDeathwhisper_TB_BaconShop_HERO_Deathwhisper ||
			cardId === CardIds.Kelthuzad_TB_BaconShop_HERO_KelThuzad
		);
	}

	static IsBaconBartender(cardId: string | null): boolean {
		return cardId?.startsWith(CardIds.BartenderBob) ?? false;
	}

	static IsBaconEnchantment(cardId: string): boolean {
		return (
			cardId === CardIds.BaconphheroHeroic ||
			cardId === CardIds.TagtransferplayerenchantDntEnchantment_Bacon_TagTransferPlayerE
		);
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Events/Parsers/Utils/BgsUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HearthstoneReplays.Events.Parsers.Utils
{
    internal class BgsUtils
    {
        public static bool IsBaconGhost(string cardId)
        {
            return cardId == CardIds.LadyDeathwhisper_TB_BaconShop_HERO_Deathwhisper
                || cardId == CardIds.Kelthuzad_TB_BaconShop_HERO_KelThuzad;
        }

        public static bool IsBaconBartender(string cardId)
        {
            return cardId?.StartsWith(CardIds.BartenderBob) ?? false;
        }

        public static bool IsBaconEnchantment(string cardId)
        {
            return cardId == CardIds.BaconphheroHeroic || cardId == CardIds.TagtransferplayerenchantDntEnchantment_Bacon_TagTransferPlayerE;
        }
    }
}
